Library/Homebrew/test/test_cmd_testbot.rb

```diff
@@ -0,0 +1,44 @@
+require "testing_env"
+require "dev-cmd/test-bot"
+
+class TestbotCommandTests < Homebrew::TestCase
+  def with_environment(h)
+    old = Hash[h.keys.map { |k| [k, ENV[k]] }]
+    ENV.update h
+    begin
+      yield
+    ensure
+      ENV.update old
+    end
+  end
+
+  def test_resolve_test_tap
+    tap = Homebrew::resolve_test_tap
+    assert_equal tap, nil, "Should return nil if no tap slug provided"
+
+    slug = "spam/homebrew-eggs"
+    url = "https://github.com/#{slug}.git"
+    pairs = [
+      {"TRAVIS_REPO_SLUG" => slug},
+      {"UPSTREAM_BOT_PARAMS" => "--tap=#{slug}"},
```

Comment:
That `--tap=spam/homebrew-eggs` is recognized is more of a convenience (no need to adjust the Travis CI repository slug beforehand), but I'd actually expect this to test `--tap=spam/eggs` or possibly both.